index.js

```diff
@@ -53,14 +53,19 @@ function callCumulusMessageAdapter(command, input) {
 }
 
 /**
- * If a Cumulus Remote Message is passed, fetch it and return a full Cumulus Message
+ * If a Cumulus Remote Message is passed, fetch it and return a full Cumulus Message with updated task metadata.
  *
  * @param {Object} cumulusMessage - either a full Cumulus Message or a Cumulus Remote Message
+ * @param {Object} context - an AWS Lambda context
  * @param {string} schemaLocations - contains location of schema files, can be null
  * @returns {Promise.<Object>} - a full Cumulus Message
  */
-function loadRemoteEvent(cumulusMessage, schemaLocations) {
-  return callCumulusMessageAdapter('loadRemoteEvent', { event: cumulusMessage, schemas: schemaLocations });
+function loadAndUpdateRemoteEvent(cumulusMessage, context, schemaLocations) {
+  return callCumulusMessageAdapter('loadAndUpdateRemoteEvent', {
+    event: cumulusMessage,
+    context,
```

Comment:
Why does the Remote event need context? How is it different from the Nested event, which already is getting the context? Just trying to understand what you're doing here.

Reply:
This change is to store the task metadata from the context, i.e. name/version/arn, in the Cumulus message. loadRemoteEvent returns the full message, so updating it there immediately is the best way to ensure that that data is captured should, say, the TaskFunction error out. loadNestedEvent's output is discarded except for `.messageConfig`, so it is not useful in this regard.